pkg/controller/batchrelease/labelpatch/patcher.go

```diff
@@ -124,7 +124,9 @@ func (r *realPatcher) patchPodBatchLabel(pods []*corev1.Pod, ctx *batchcontext.B
 			klog.InfoS("Pod batchID is not a number, skip patching", "pod", klog.KObj(pod), "rollout", r.logKey)
 			continue
 		}
-		plannedUpdatedReplicasForBatches[podBatchID-1]--
+		if podBatchID > 0 && podBatchID <= len(plannedUpdatedReplicasForBatches) {
```

Comment:
Is it convenient to add UT and annotations? Can you make it easier to add UT and comments?